src/ui/Overlay.js

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react';
+import styles from './styles/overlay.module.css';
+import CharacterSheet from './sheet/CharacterSheet';
+import ProfileSelection from './profile/ProfileSelection';
+import NestedTab from './tabs/NestedTab';
+
+function Overlay() {
+  const [activeTab, setActiveTab] = useState('Select Character');
+  const [sheet, setSheet] = useState(null);
+
+  const onCharacterReady = (characterSheetPromise) => {
+    characterSheetPromise.then((sheetData) => {
+      setActiveTab('Character Sheet');
+      setSheet(sheetData);
+    });
+  };
+
+  return (
+    <div className={`globalOverlay ${styles.Overlay}`}>
+      <NestedTab title="Select Character" onClick={() => setActiveTab('Select Character')} active>
+        <ProfileSelection onCharacterReady={onCharacterReady} />
+        <NestedTab title="Character Sheet" active={activeTab === 'Character Sheet'}>
+          <CharacterSheet sheet={sheet} />
+        </NestedTab>
+      </NestedTab>
+    </div>
+  );
+}
+
+export default Overlay;
```

Comment:
You can combine `export`s with the definition:

```js
export default function Overlay() { ...
```

Reply:
It makes it easier to read a module because it's immediately apparent what is and isn't exported.

Reply:
This seems like a style preference? I'm probably biased by how I've been doing it, but the current style makes more sense to me in terms of seeing what is and isn't exported

Reply:
I suppose it is a preference. I actually figured you would prefer it inline like this because that's the way that Rust does it.

The way I see it, it's similar to classes, where each member has its access modifier right there next to it.

Reply:
With `export default` there's only one, so I'd prefer to keep it consistent across all modules by putting it at the bottom. Also I'm being lazy and don't want to do extra changes 😄